docker/Dockerfile

```diff
@@ -66,10 +71,11 @@ RUN \
 	# Remove Git and clean apt cache
 	apt-get clean && \
 	apt-get remove -y git && \
-	apt-get -y autoremove
+	apt-get -y autoremove && \ 
+	chmod +x docker/entrypoint.sh
```

Comment:
Looking to Cygnus Dockerfile (https://github.com/telefonicaid/fiware-cygnus/blob/master/docker/cygnus-ngsi/Dockerfile) we don't need to chmod +x the entrypoint file there. Why this is needed here?

Reply:
@fgalan As suggested in comment https://github.com/telefonicaid/iotagent-json/pull/380#discussion_r255063951 permission was added. What do you think ?

Reply:
When we run entrypoint.sh script in our local its asks for permission so we decided to add it in the Dockerfile itself.

Reply:
I understand that is you set the permission in the file itself in the repository you don't need to use chmod +x in the Dockerfile.

Note `ls` run in the repository working copy shows +x for that file

```
$ ls -l
total 88
-rw-r--r-- 1 fermin fermin 11366 feb 20 12:37 agent.conf
-rw-r--r-- 1 fermin fermin   159 sep 13  2017 cartodb_keys.conf
-rwxr-xr-x 1 fermin fermin 50304 dic 13 10:46 cygnus-entrypoint.sh
-rw-r--r-- 1 fermin fermin 12489 feb 20 12:37 Dockerfile
-rw-r--r-- 1 fermin fermin  1016 oct  9 18:51 README.md
```

This seems the way in which it is used in Cygnus, so please do the same.

Reply:
@fgalan :

> we don't need to chmod +x the entrypoint file there. Why this is needed here?

The IoT Agents set a `USER` , cygnus less secure as it will be running using the default root user.

```Dockerfile
USER node
````

Maybe someone should raise this as a security issue on Cygnus?

The advantage of leaving in the `chmod` statement is that it is explicit. This way all files can be left as `-rw` by default and you're not relying on some access permissions flags. These permissions flags are hidden from the developer by default.

Reply:
@fgalan So the final conclusion is to add chmod +x I guess ?

Reply:
Ok. No so important. NTC

However, I think is a common practice (maybe a bad habit? :) to run processes *inside* docker containers as root. However, if you feel this could be a security risk, maybe a PR fixing it (i.e. adding `USER` to the several Cygnus Dockerfiles plus usage of chmod +x on the entrypoint file) would be a good contribution.

Reply:
The **principle of least privilege** applies here - *especially* regarding passwords.

* https://medium.com/@mccode/processes-in-containers-should-not-run-as-root-2feae3f0df3b